subsys/dfu/boot/mcuboot.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2016 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCUBOOT_H__
+#define __MCUBOOT_H__
+
+#include "boot/mcuboot_constraints.h"
+
+/**
+ * @brief Marks the image in slot 0 as confirmed. The system will continue booting
+ * into the image in slot 0 until told to boot from a different slot.
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int boot_img_confirmed_set(void);
+
+/**
+ * @brief Marks the image in slot 1 as pending. On the next reboot, the system will
+ * perform a one-time boot of the slot 1 image.
+ *
+ * @param permanent Whether the image should be used permanently or
+ * only tested once:
+ *   0=run image once, then confirm or revert.
+ *   1=run image forever.
+ * @return 0 on success, negative errno code on fail.
+ */
+int boot_img_pending_set(int permanent);
```

Comment:
This doesn't quite make sense. The operation we should have is to mark slot 1 for upgrade, with a permanent flag. It seems like there might be a better name for that, "boot_request_upgrade" perhaps?